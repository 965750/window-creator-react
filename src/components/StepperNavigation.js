import React, { Component } from 'react'
import { connect } from 'react-redux'
import { changeStep } from '../store/actions/themeActions'

class StepperNavigation extends Component {
    render() {
        let btnBack

        if (this.props.currentStep !== 1) {
            btnBack = <button className='h-8 mr-4 uppercase border-2 border-gullGray w-full cursor-pointer' onClick={() => this.props.handleChangeStep(this.props.currentStep - 1)}>
            back
        </button>            
        }
        return (
            <div className='flex absolute w-full bottom-0'>
                { btnBack }
                <button className='h-8 min-w-half uppercase border-2 bg-gullGray text-white border-gullGray cursor-pointer' onClick={() => this.props.handleChangeStep(this.props.currentStep + 1)}>
                    next step
                </button>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        currentStep: state.theme.stepper.active
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        handleChangeStep: (step) => dispatch(changeStep(step))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(StepperNavigation)
