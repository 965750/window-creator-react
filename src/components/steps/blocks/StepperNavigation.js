import React, { Component } from 'react'
import { connect } from 'react-redux'
import { changeStep } from '../../../store/actions/themeActions'
import { FormattedMessage } from 'react-intl'
import PropTypes from 'prop-types'

class StepperNavigation extends Component {
    render() {
        let btnBack

        if (this.props.activeStep !== 1) {
            btnBack = (
                <button
                    className="h-8 mr-4 uppercase border-2 border-gullGray w-full cursor-pointer"
                    onClick={() =>
                        this.props.handleChangeStep(this.props.activeStep - 1)
                    }
                >
                    <FormattedMessage id="back" defaultMessage="back" />
                </button>
            )
        }
        return (
            <div className="flex absolute w-full bottom-0">
                {btnBack}
                <button
                    className="h-8 min-w-half uppercase border-2 bg-gullGray text-white border-gullGray cursor-pointer"
                    onClick={() =>
                        this.props.handleChangeStep(this.props.activeStep + 1)
                    }
                >
                    <FormattedMessage
                        id="next step"
                        defaultMessage="next step"
                    />
                </button>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        activeStep: state.theme.stepper.active,
    }
}

const mapDispatchToProps = dispatch => {
    return {
        handleChangeStep: step => dispatch(changeStep(step)),
    }
}

StepperNavigation.propTypes = {
    activeStep: PropTypes.number.isRequired,
    handleChangeStep: PropTypes.func,
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(StepperNavigation)
