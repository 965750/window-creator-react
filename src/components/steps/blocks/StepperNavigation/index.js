import { changeStep } from '../../../../store/actions/themeActions'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'

import PropTypes from 'prop-types'
import React from 'react'

const StepperNavigation = ({ activeStep, handleChangeStep }) => {
  let btnBack
  let btnNext

  if (activeStep !== 1) {
    btnBack = (
      <button
        data-testid="prevStepBtn"
        className="h-8 mr-1 uppercase border-2 border-gullGray w-1/2 cursor-pointer"
        onClick={() => handleChangeStep(activeStep - 1)}
      >
        <FormattedMessage id="back" defaultMessage="back" />
      </button>
    )
  }

  if (activeStep < 3) {
    btnNext = (
      <button
        data-testid="nextStepBtn"
        className="h-8 w-1/2 uppercase border-2 bg-gullGray text-white border-gullGray cursor-pointer"
        onClick={() => handleChangeStep(activeStep + 1)}
      >
        <FormattedMessage id="next step" defaultMessage="next step" />
      </button>
    )
  }

  return (
    <div className="flex w-full">
      {btnBack}
      {btnNext}
    </div>
  )
}

const mapStateToProps = (state) => ({
  activeStep: state.theme.stepper.active,
})

const mapDispatchToProps = (dispatch) => ({
  handleChangeStep: (step) => dispatch(changeStep(step)),
})

StepperNavigation.propTypes = {
  activeStep: PropTypes.number.isRequired,
  handleChangeStep: PropTypes.func,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(StepperNavigation)
