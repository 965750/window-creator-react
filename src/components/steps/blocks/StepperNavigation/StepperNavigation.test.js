import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup, fireEvent } from '@testing-library/react';

import StepperNavigation from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
    stepper: {
      steps: [
        {
          label: 'Step 1',
          title: 'Choose Door',
          id: 1,
        },
        {
          label: 'Step 2',
          title: 'Choose Door',
          id: 2,
        },
      ],
      active: 2
    }
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    case 'CHANGE_STEP':
      return {
        ...state,
        theme: {
          stepper: {
            active: action.step
          }
        }
      }
    case 'RESET_STORE':
      return initialStore
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('StepperNavigation', () => {
  let wrapper
  const initialProps = {}

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <StepperNavigation {...props} />
      </IntlProvider>,
    )
  }

  afterEach(() => {
    cleanup

    store.dispatch({
      type: 'RESET_STORE',
    })
  })

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should increment step by 1 on button click', () => {
    setUp()

    expect(store.getState().theme.stepper.active).toBe(2)

    fireEvent.click(wrapper.queryByTestId('nextStepBtn'))

    expect(store.getState().theme.stepper.active).toBe(3)
  })

  it('should decrement step by 1 on button click', () => {
    setUp()

    expect(store.getState().theme.stepper.active).toBe(2)

    fireEvent.click(wrapper.queryByTestId('prevStepBtn'))

    expect(store.getState().theme.stepper.active).toBe(1)
  })

  it('should NOT show prevStepBtn when first step is active', () => {
    setUp()

    expect(store.getState().theme.stepper.active).toBe(2)

    fireEvent.click(wrapper.queryByTestId('prevStepBtn'))

    expect(wrapper.queryByTestId('prevStepBtn')).toBeNull()
  })

  it('should NOT show nextStepBtn when last step is active', () => {
    setUp()

    expect(store.getState().theme.stepper.active).toBe(2)

    fireEvent.click(wrapper.queryByTestId('nextStepBtn'))

    expect(wrapper.queryByTestId('nextStepBtn')).toBeNull()
  })
})
