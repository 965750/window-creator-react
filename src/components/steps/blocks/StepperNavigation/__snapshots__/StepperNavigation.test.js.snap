// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepperNavigation should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex w-full"
  >
    <button
      class="h-8 mr-1 uppercase border-2 border-gullGray w-1/2 cursor-pointer"
      data-testid="prevStepBtn"
    >
      powrót
    </button>
    <button
      class="h-8 w-1/2 uppercase border-2 bg-gullGray text-white border-gullGray cursor-pointer"
      data-testid="nextStepBtn"
    >
      następny krok
    </button>
  </div>
</DocumentFragment>
`;
